Export macro picks up local `Result` type

While migrating to the latest master, I noticed a tiny problem which looks like a regression. Previously in 0.9.x it was possible to have a local `Result` type in scope. On master this fails to compile:

```rust
use gdnative::prelude::*;

// This line is the only modification of the hello world example:
struct Result;

#[derive(NativeClass)]
#[inherit(Node)]
struct HelloWorld;

#[methods]
impl HelloWorld {
    fn new(_owner: &Node) -> Self {
        HelloWorld
    }

    #[export]
    fn _ready(&self, _owner: &Node) {
        godot_print!("hello, world.")
    }
}

fn init(handle: InitHandle) {
    handle.add_class::<HelloWorld>();
}

godot_init!(init);
```

The compilation fails in the exported function, because the macro now seems to pick up the local `Result` but probably wants to use some `Result` from the `gdnative` crate:

```
error[E0107]: this struct takes 0 generic arguments but 2 generic arguments were supplied
  --> src/lib.rs:16:8
   |
16 |     fn _ready(&self, _owner: &Node) {
   |        ^^^^^^
   |        |
   |        expected 0 generic arguments
   |        help: remove these generics
   |
note: struct defined here, with 0 generic parameters
  --> src/lib.rs:3:8
   |
3  | struct Result;
   |        ^^^^^^
   = note: this error originates in the derive macro `FromVarargs` (in Nightly builds, run with -Z macro-backtrace for more info)
```

Most likely this is straightforward to fix by some `$crate` usage in the macro, right?

--- a/bindings_generator/src/methods.rs
+++ b/bindings_generator/src/methods.rs
@@ -218,7 +218,7 @@ pub fn generate_method_table(api: &Api, class: &GodotClass) -> TokenStream {
             pub fn get(gd_api: &GodotApi) -> &'static Self {
                 unsafe {
                     let table = Self::get_mut();
-                    static INIT: ::std::sync::Once = Once::new();
+                    static INIT: std::sync::Once = std::sync::Once::new();
                     INIT.call_once(|| {
                         #method_table::init(table, gd_api);
                     });
@@ -618,7 +618,7 @@ mod varargs_call {
         let arg_count = sig.arguments.len();
         let method_body = {
             let args_buffer = quote! {
-                let mut argument_buffer: Vec<*const sys::godot_variant> = Vec::with_capacity(#arg_count + varargs.len());
+                let mut argument_buffer: std::vec::Vec<*const sys::godot_variant> = std::vec::Vec::with_capacity(#arg_count + varargs.len());
             };
 
             let args = arguments.clone().map(|(name, ty)| {

--- a/gdnative-bindings/src/generated.rs
+++ b/gdnative-bindings/src/generated.rs
@@ -4,7 +4,6 @@ use libc;
 use libc::c_char;
 use std::mem;
 use std::ptr;
-use std::sync::Once;
 
 use gdnative_core::core_types::*;
 use gdnative_core::object::*;

--- a/gdnative-derive/src/varargs.rs
+++ b/gdnative-derive/src/varargs.rs
@@ -38,8 +38,8 @@ pub(crate) fn derive_from_varargs(input: DeriveInput) -> Result<TokenStream2, sy
                     impl #generics ::gdnative::export::FromVarargs for #ident #generics #where_clause {
                         fn read<'a>(
                             #input_ident: &mut ::gdnative::export::Varargs<'a>,
-                        ) -> Result<Self, Vec<::gdnative::export::ArgumentError<'a>>> {
-                            Ok(#ident)
+                        ) -> std::result::Result<Self, std::vec::Vec<::gdnative::export::ArgumentError<'a>>> {
+                            std::result::Result::Ok(#ident)
                         }
                     }
                 })
@@ -116,8 +116,8 @@ pub(crate) fn derive_from_varargs(input: DeriveInput) -> Result<TokenStream2, sy
             impl #generics ::gdnative::export::FromVarargs for #ident #generics #where_clause {
                 fn read<'a>(
                     #input_ident: &mut ::gdnative::export::Varargs<'a>,
-                ) -> Result<Self, Vec<::gdnative::export::ArgumentError<'a>>> {
-                    let mut __errors = Vec::new();
+                ) -> std::result::Result<Self,std::vec:: Vec<::gdnative::export::ArgumentError<'a>>> {
+                    let mut __errors = std::vec::Vec::new();
 
                     #(
                         let #req_var_idents = #input_ident.read()
@@ -140,14 +140,14 @@ pub(crate) fn derive_from_varargs(input: DeriveInput) -> Result<TokenStream2, sy
                     )*
 
                     if !__errors.is_empty() {
-                        return Err(__errors);
+                        return std::result::Result::Err(__errors);
                     }
 
                     #(
                         let #req_var_idents = #req_var_idents.unwrap();
                     )*
 
-                    Ok(#ident {
+                    std::result::Result::Ok(#ident {
                         #(#req_var_idents,)*
                         #(#opt_var_idents,)*
                     })

--- a/gdnative-derive/src/variant/from.rs
+++ b/gdnative-derive/src/variant/from.rs
@@ -65,14 +65,14 @@ pub(crate) fn expand_from_variant(derive_data: DeriveData) -> Result<TokenStream
                     let __dict = ::gdnative::core_types::Dictionary::from_variant(#input_ident)
                         .map_err(|__err| FVE::InvalidEnumRepr {
                             expected: VariantEnumRepr::ExternallyTagged,
-                            error: Box::new(__err),
+                            error: std::boxed::Box::new(__err),
                         })?;
 
                     let __keys = __dict.keys();
                     if __keys.len() != 1 {
                         Err(FVE::InvalidEnumRepr {
                             expected: VariantEnumRepr::ExternallyTagged,
-                            error: Box::new(FVE::InvalidLength {
+                            error: std::boxed::Box::new(FVE::InvalidLength {
                                 expected: 1,
                                 len: __keys.len() as usize,
                             }),
@@ -82,7 +82,7 @@ pub(crate) fn expand_from_variant(derive_data: DeriveData) -> Result<TokenStream
                         let __key = String::from_variant(&__keys.get(0))
                             .map_err(|__err| FVE::InvalidEnumRepr {
                                 expected: VariantEnumRepr::ExternallyTagged,
-                                error: Box::new(__err),
+                                error: std::boxed::Box::new(__err),
                             })?;
                         match __key.as_str() {
                             #(
@@ -90,7 +90,7 @@ pub(crate) fn expand_from_variant(derive_data: DeriveData) -> Result<TokenStream
                                     let #var_input_ident_iter = &__dict.get_or_nil(&__keys.get(0));
                                     (#var_from_variants).map_err(|err| FVE::InvalidEnumVariant {
                                         variant: #ref_var_ident_string_literals,
-                                        error: Box::new(err),
+                                        error: std::boxed::Box::new(err),
                                     })
                                 },
                             )*

--- a/gdnative-derive/src/variant/repr.rs
+++ b/gdnative-derive/src/variant/repr.rs
@@ -231,7 +231,7 @@ impl VariantRepr {
                             ::gdnative::core_types::VariantArray::from_variant(#variant)
                                 .map_err(|__err| FVE::InvalidStructRepr {
                                     expected: VariantStructRepr::Tuple,
-                                    error: Box::new(__err),
+                                    error: std::boxed::Box::new(__err),
                                 })
                                 .and_then(|__array| {
                                     let __expected = #expected_len;
@@ -245,11 +245,11 @@ impl VariantRepr {
                                             let #non_skipped_idents = #non_skipped_exprs
                                                 .map_err(|err| FVE::InvalidItem {
                                                     index: __index as usize,
-                                                    error: Box::new(err),
+                                                    error: std::boxed::Box::new(err),
                                                 })?;
                                         )*
                                         #(
-                                            let #skipped_idents = ::std::default::Default::default();
+                                            let #skipped_idents = std::default::Default::default();
                                         )*
                                         Ok(#ctor( #(#ctor_idents),* ))
                                     }
@@ -290,7 +290,7 @@ impl VariantRepr {
                         ::gdnative::core_types::Dictionary::from_variant(#variant)
                             .map_err(|__err| FVE::InvalidStructRepr {
                                 expected: VariantStructRepr::Struct,
-                                error: Box::new(__err),
+                                error: std::boxed::Box::new(__err),
                             })
                             .and_then(|__dict| {
                                 #(
@@ -299,11 +299,11 @@ impl VariantRepr {
                                     let #non_skipped_idents = #exprs
                                         .map_err(|err| FVE::InvalidField {
                                             field_name: __field_name,
-                                            error: Box::new(err),
+                                            error: std::boxed::Box::new(err),
                                         })?;
                                 )*
                                 #(
-                                    let #skipped_idents = ::std::default::Default::default();
+                                    let #skipped_idents = std::default::Default::default();
                                 )*
                                 Ok(#ctor { #( #ctor_idents ),* })
                             })

--- a/gdnative-sys/build.rs
+++ b/gdnative-sys/build.rs
@@ -523,10 +523,10 @@ mod api_wrapper {
             }
         }
         quote! {
-            pub unsafe fn from_raw(core_api_struct: *const godot_gdnative_core_api_struct) -> Result<Self, InitError> {
+            pub unsafe fn from_raw(core_api_struct: *const godot_gdnative_core_api_struct) -> std::result::Result<Self, InitError> {
                 #godot_apis
                 #struct_field_bindings
-                Ok(GodotApi{
+                std::result::Result::Ok(GodotApi{
                     #constructed_struct_fields
                 })
             }

--- a/impl/proc_macros/src/pool_array_element.rs
+++ b/impl/proc_macros/src/pool_array_element.rs
@@ -390,13 +390,13 @@ pub fn decl_element(input: proc_macro::TokenStream) -> Result<TokenStream, syn::
         #(#attrs)*
         pub trait #trait_name: private::Sealed {
             #[doc(hidden)]
-            type SysArray: Default;
+            type SysArray: std::default::Default;
             #[doc(hidden)]
             type SysReadAccess;
             #[doc(hidden)]
             type SysWriteAccess;
             #[doc(hidden)]
-            type SysTy: Default;
+            type SysTy: std::default::Default;
             #[doc(hidden)]
             type SysRefTy;
 

